test/adapter/json_api/top_level_links_test.rb

```diff
@@ -0,0 +1,101 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class Adapter
+      class JsonApi
+        class TopLevelLinksTest < Minitest::Test
+          URI = 'http://example.com'
+
+          def setup
+            ActionController::Base.cache_store.clear
+            @blog = Blog.new(id: 1,
+                             name: 'AMS Hints',
+                             writer: Author.new(id: 2, name: "Steve"),
+                             articles: [Post.new(id: 3, title: "AMS")])
+          end
+
+          def load_adapter(paginated_collection, options = {})
+            options = options.merge(adapter: :json_api)
+            ActiveModel::SerializableResource.new(paginated_collection, options)
+          end
+
+          def test_links_is_not_present_when_not_defined
+            adapter = load_adapter(@blog)
+
+            expected = {
+              :data => {
+                :id => "1",
+                :type => "blogs",
+                :attributes => {
+                  :name => "AMS Hints"
+                },
+                :relationships => {
+                  :writer=> {:data => {:type => "authors", :id => "2"}},
+                  :articles => {:data => [{:type => "posts", :id => "3"}]}
+               }
+            }}
```

Comment:
According to the spec, non-available links should be `nil` or omitted. So an empty `links` attribute means "no links available" and not "no links provided".